source/retryable-writes/retryable-writes.rst

```diff
@@ -511,6 +513,15 @@ experience across all drivers.
 Why expose users to server-side errors for unsupported operations?
 ------------------------------------------------------------------
 
+.. todo:: Determine how this should change. Do we end up using whitelists for
+   supported operations (based on wire protocol verison) and avoid injecting
+   transaction IDs on all write ops, or rely on the server to raise an error if
+   unsupported ops would be retried? In both cases, it's possible for users to
+   be unaware that some operations are not supported until some error in prod.
+   Given the MongoClient-wide scope of retryWrites, we likely don't want to
+   prohibit unsupported operations outright (either in the client API or by
+   having the server raise an exception on the first attempt).
+
```

Comment:
Let's discuss how we'd like to handle unsupported operations. This was an outstanding question following the design meetings earlier in the week.

Reply:
This seems like an easy problem to solve, don't retry unacknowledged writes and document that we don't retry unacknowledged writes and why.

Reply:
Bernie's comment *does* offer a simple solution for unacknowledged writes.

As for acknowledged writes. "Having the server only return an error on retry attempts" isn't even possible. When the server receives a message from the client for a non-retryable operation, the server doesn't know whether this is the first or second attempt at all. It can't return an error on the second attempt, therefore.

Jeremy I agree that the two options you listed are the two options available to us. Let's just ask Andy and Eliot which they want?

Reply:
@rstam, @jyemin, @ajdavis, and I spoke with @andy10gen earlier and concluded that drivers should implement a whitelist of commands that support retriability. I will update the spec accordingly and add an explanation of the design decision to the spec as well.

Reply:
https://github.com/mongodb/specifications/pull/174/commits/8eb5d59b8309c9be85142b5f0e6c2f923d94a85a changes the spec to utilize a whitelist of supported operations